pkg/controllers/corecluster/corecluster_controller.go

```diff
@@ -0,0 +1,242 @@
+			builder.WithPredicates(coreClusterPredicate(r.ManagedNamespace)),
+		).
+		Complete(r); err != nil {
+		return fmt.Errorf("failed to create controller: %w", err)
+	}
+
+	return nil
+}
+
+// Reconcile reconciles the core cluster object for the openshift-cluster-api namespace.
+func (r *CoreClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithName(controllerName)
+	logger.Info("Reconciling core cluster")
+	defer logger.Info("Finished reconciling core cluster")
+
+	ocpInfrastructureName, err := getOCPInfrastructureName(r.Infra)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to obtain ocp infrastructure name: %w", err)
+	}
+
+	cluster, err := r.ensureCoreCluster(ctx, client.ObjectKey{Namespace: r.ManagedNamespace, Name: ocpInfrastructureName}, logger)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to ensure core cluster: %w", err)
+	}
+
+	if !cluster.DeletionTimestamp.IsZero() {
+		if err := r.SetStatusAvailable(ctx, ""); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to set status available: %w", err)
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.ensureCoreClusterControlPlaneInitializedCondition(ctx, cluster); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to ensure core cluster has the ControlPlaneInitializedCondition: %w", err)
+	}
+
+	if err := r.SetStatusAvailable(ctx, ""); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to set status available: %w", err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// ensureCoreCluster creates a cluster with the given name and returns the cluster object.
+func (r *CoreClusterReconciler) ensureCoreCluster(ctx context.Context, clusterObjectKey client.ObjectKey, logger logr.Logger) (*clusterv1.Cluster, error) {
+	cluster := &clusterv1.Cluster{}
+	if err := r.Client.Get(ctx, clusterObjectKey, cluster); err != nil && !kerrors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get core cluster %s/%s: %w", clusterObjectKey.Namespace, clusterObjectKey.Name, err)
+	} else if err == nil {
+		return cluster, nil
+	}
+
+	if r.Infra.Status.PlatformStatus == nil {
+		return nil, errPlatformStatusShouldNotBeNil
+	}
+
+	infraClusterKind, infraClusterAPIVersion, err := mapOCPPlatformToInfraClusterKindAndVersion(r.Platform)
+	if err != nil {
+		return nil, fmt.Errorf("unable to map Infrastucture resource platform type to Cluster API infra cluster kind: %w", err)
+	}
+
+	infraCluster := &unstructured.Unstructured{}
+	infraCluster.SetKind(infraClusterKind)
+	infraCluster.SetAPIVersion(infraClusterAPIVersion)
+
+	if err := r.Client.Get(ctx, clusterObjectKey, infraCluster); err != nil {
+		return nil, fmt.Errorf("failed to get infra cluster %s/%s: %w", clusterObjectKey.Namespace, clusterObjectKey.Name, err)
+	}
+
+	logger.Info(fmt.Sprintf("core cluster %s/%s does not exist, creating it", clusterObjectKey.Namespace, clusterObjectKey.Name))
```

Comment:
Other info logs are being capitalised
```suggestion
	logger.Info(fmt.Sprintf("Core cluster %s/%s does not exist, creating it", clusterObjectKey.Namespace, clusterObjectKey.Name))
```